src/shared_modules/content_manager/src/components/CtiApiDownloader.hpp

```diff
@@ -121,28 +153,58 @@ class CtiApiDownloader final : public AbstractHandler<std::shared_ptr<UpdaterCon
      */
     void downloadContent(int toOffset, const std::string& fullFilePath) const
     {
+        // Define the parameters for the request.
+        const auto queryParameters = "/changes?from_offset=" + std::to_string(m_context->currentOffset) +
+                                     "&to_offset=" + std::to_string(toOffset);
+
+        // On download success routine.
         const auto onSuccess {[]([[maybe_unused]] const std::string& data)
                               {
                                   std::cout << "CtiApiDownloader - Request processed successfully.\n";
                               }};
 
+        // On download error routine.
         const auto onError {
-            [this](const std::string& message, [[maybe_unused]] const long statusCode)
+            [](const std::string& message, const long statusCode)
             {
-                // Set the status of the stage
-                m_context->data.at("stageStatus").push_back(R"({"stage": "CtiApiDownloader", "status": "fail"})"_json);
+                const std::string exceptionMessage {"Error " + std::to_string(statusCode) + " from server: " + message};
 
-                throw std::runtime_error("CtiApiDownloader - Could not get response from API because: " + message);
+                // If there is an error from the server, throw a different exception.
+                if (statusCode >= 500 && statusCode <= 599)
+                {
+                    throw cti_server_error {exceptionMessage};
+                }
+                throw std::runtime_error {exceptionMessage};
             }};
 
-        const auto fromOffset = m_context->currentOffset;
+        // Loop for retrying the downloads from the server until the download is successful or there is an HTTP error
+        // different from 5xx.
+        constexpr auto INITIAL_SLEEP_TIME {1};
+        auto sleepTime {INITIAL_SLEEP_TIME};
+        auto retryAttempt {1};
+        auto retry {true};
+        while (retry)
+        {
+            try
+            {
+                m_urlRequest.get(HttpURL(m_url + queryParameters), onSuccess, onError, fullFilePath);
+                retry = false;
+            }
+            catch (cti_server_error e)
```

Comment:
improve: Catch handler catches by value; should catch by reference instead

```suggestion
            catch (cti_server_error& e)
```

Reply:
Improve: unused exception parameter 'e'

```suggestion
            catch ([[maybe_unused]] const std::exception& e)
```

Reply:
> improve: Catch handler catches by value; should catch by reference instead

Done.